common/Geometry2d/Segment.cpp

```diff
@@ -18,6 +18,13 @@ Rect Segment::bbox() const {
 }
 
 float Segment::distTo(const Point& other) const {
+    float dist = nearestPoint(other).distTo(other);
+    if (nearlyEqual(dist, 0)) {
```

Comment:
Its assumed by some other functions that distTo()==0 means the point is on the line. On the otherhand I could make an actual onLine function.